notification/twilio/voice.go

```diff
@@ -638,3 +597,56 @@ func (v *Voice) FriendlyValue(ctx context.Context, value string) (string, error)
 	}
 	return libphonenumber.Format(num, libphonenumber.INTERNATIONAL), nil
 }
+
+// buildMessage is a function that will build the VoiceOptions object with the proper message contents
+func buildMessage(prefix string, msg notification.Message) (opts *VoiceOptions, err error) {
```

Comment:
While having this logic moved out of `Send` makes it a bit easier to test, we still have a nested structure in and out along with the base64 encoding of the message AND rendering said message all in one place. That makes the test cases bulky and because they need to validate multiple concerns, hard to read/maintain.

If we break it into some smaller pieces around each concern, it will be easier to write (and, more importantly, read) test cases for them.

Let's try:
- one func, `buildMessage`, that takes the app name & `notification.Message` and outputs a string
- one method, `VoiceOptions.setMsgBody`, to handle encoding the message returned from `buildMessage` (this will be expanded in a future PR for the pause work)
- one method, `VoiceOptions.setMsgParams`, to set the remaining parameters given a `notification.Message`


```go
// buildMessage will generate a voice call message for the provided notification.Message.
func buildMessage(appName string, msg notification.Message) (string, error) {
	// ...
}

// setMsgBody will encode and set/update the message body parameter.
func (vo *VoiceOptions) setMsgBody(body string) {
	if vo.Params == nil {
		vo.Params = make(url.Values)
	}
	// Encode the body, so we don't need to worry about
	// buggy apps not escaping URL params properly.
	vo.Params.Set(msgParamBody, b64enc.EncodeToString([]byte(body)))
}

// setMsgParams will set parameters for the provided message.
func (vo *VoiceOptions) setMsgParams(n notification.Message) error {
	// ...
}
```

The `Send` function will then have something like:
```go
func (v *Voice) Send(// ...	
	opts := &VoiceOptions{
		ValidityPeriod: time.Second * 10,
	}

	err = opts.setMsgParams(msg)
	if err != nil {
		return err
	}

	msgBody, err := buildMessage(cfg.ApplicationName(), msg)
	if err != nil {
		return err
	}
	opts.setMsgBody(msgBody)
```
---
We'll be sacrificing the single switch statement for two. Still, I think it's necessary to have one testable function that takes in a `notification.Message` and outputs spoken text that's easy to validate.

The voice-equivalent of `renderAlertMessage` for SMS.